sources/packages/backend/libs/services/src/notifications/notification/notification-actions.service.ts

```diff
@@ -128,6 +129,51 @@ export class NotificationActionsService {
     );
   }
 
+  /**
+   * Creates a notification when there is a partial match for a student account.
+   * @param notification input parameters to generate the notification.
+   * @param auditUserId user that should be considered the one that is causing the changes.
+   * @param entityManager optional repository that can be provided, for instance,
+   * to execute the command as part of an existing transaction. If not provided
+   * the local repository will be used instead.
+   */
+  async savePartialStudentMatchNotification(
+    notification: PartialStudentMatchNotification,
+    auditUserId: number,
+    entityManager?: EntityManager,
+  ): Promise<void> {
+    const { templateId, emailContacts } =
+      await this.assertNotificationMessageDetails(
```

Comment:
Not part of your PR but the `entityManager` should be passed down to the private method `assertNotificationMessageDetails`. It would allow Typeorm to reuse the same transaction instead of creating a second one. The method itself is already called the final method with an optional transaction.
If you have time at the end we can have this change done for all these methods.

Reply:
Just to be clear, I am assuming that this will not be implemented as part of this effort, right?

Reply:
Correct. It's separate from this PR as you said.